Bio/PDB/mmtf/mmtfio.py

```diff
@@ -104,6 +104,8 @@ def _save_structure(self, filepath, select):
             header_dict["structure_method"] = []
         else:
             header_dict["structure_method"] = [header_dict["structure_method"]]
+        if header_dict["biomoltrans"] == "":
```

Comment:
This code would be much nicer if we used `dict.get` instead of `setdefault`, or simply 'if key not in dict: set key`

Reply:
Would you like me to change the entries above too?

Reply:
I think the following should be good:
```
        if header_dict.get("biomoltrans") in ["", None]:
            header_dict["biomoltrans"] = None
```

Reply:
I tested this with an empty dict and it worked

Reply:
However, it looks like having a defaultdict initially as str gives the empty string when given any missing key like the current code:
```
In [1]: from collections import defaultdict

In [2]: header_dict = defaultdict(str, {})

In [3]: header_dict["biomoltrans"]
Out[3]: ''
```

Reply:
I looked at this and wondered about the style, but it was consistent with the existing code so didn't flag it.

I wouldn't use a defaultdict here - that might have side effects downstream for people using the dictionary.

Reply:
Based on the tests, it's already being used in a way that needs it to be like that and people might get more unexpected effects if we change it. I think it's fine to have an empty field rather than a key error. 

There are also downstream programs like BioTite that require there to be some way of accessing every field even if it's empty so I think the original implementation makes sense here.